src/dune_rules/foreign.ml

```diff
@@ -146,6 +146,29 @@ module Stubs = struct
   let decode = Dune_lang.Decoder.fields decode_stubs
 end
 
+module Object = struct
+  type t = { name : string }
+
+  let decode =
+    let open Dune_lang.Decoder in
+    let+ name = Dune_lang.Decoder.string in
+    { name }
+
+  let suffix = ".o"
```

Comment:
You should use `ext_obj` here so that it works on different systems (`.obj` for Windows, etc).